### PR TITLE
Contributors filtering and aggregations

### PR DESCRIPTION
**Pull request by warrd**
https://github.com/wellcomecollection/catalogue/issues/1260

### CODE REVIEW

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/WorksParams.scala [hunk 1]
```diff
@@ -142,12 +146,65 @@ object MultipleWorksParams extends QueryParamsUtils {
...
+          sort,
+          sortOrder,
+          query,
+          identifiers,
+          itemLocationType,
+          itemLocationTypeId,
+          accessStatus,
+          workType,
+          partOf) =>
+        // Scala has a max tuple size of 22 so this is nested to get around this limit
```
Comment (jamieparkinson) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/WorksParams.scala:173]: lol, I noticed we were about to hit this the other day. Possibly worth a refactor into `parameterMap` at some point

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala [hunk 2]
```diff
@@ -66,6 +66,12 @@ object WorksRequestBuilder
         .field("data.subjects.label.keyword")
         .minDocCount(0)
 
+    case AggregationRequest.Contributor =>
+      TermsAggregation("contributors")
+        .size(20)
```
Comment (jamieparkinson) [inline api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala:71]: Does 20 include all the contributors we have?

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/WorksParams.scala (hunk 1 above)
Comment (jamieparkinson) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/WorksParams.scala:173]: fwiw the tuple unpacking is slightly nicer in the latest version of akka-http which I've upgraded to in this PR https://github.com/wellcomecollection/catalogue/pull/1230 that I need to get round to merging and applying

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala (hunk 2 above)
Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala:71]: I chose this the same as `subject` and `genre`. I have no idea tbh. Maybe @jamesgorrie has some idea?

**Review by jamieparkinson — APPROVED** [review]

File: api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala (hunk 2 above)
Comment (warrd) [reply api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala:71]: For some context on number of aggregations, this comment:
https://github.com/wellcomecollection/platform/issues/3944#issuecomment-547065812

**Pull request merged**